### ISSUE TITLE
SemiBold variant

### ISSUE THREAD

**Issue by cpitclaudel** [daltonmaag/ubuntu#3]
Hi there,

I use Ubuntu Mono for virtually everything programming-related that I do, and often find myself wanting a semibold or medium variant (typical use case: highlighting argument names in docstrings, where I find full bold is too prominent).  

I tried interpolating between Regular and Bold using MutatorMath (I described my efforts at https://github.com/LettError/MutatorMath/issues/58), but that didn't work too well: the UFOs that I derived from the released TTFs weren't interpolatable. I tried it with the UFOs from this repository as well, and it seemed to work better, but the glyphs were still entirely distorted (which makes sense, now that I understand the stuff about interpolation requiring matching points and angles)

![screenshot from 2016-10-25 02-30-46](https://cloud.githubusercontent.com/assets/8181630/19675213/1169d798-9a5b-11e6-88b2-dff169707752.png)

Bottom line: a semibold variant would be great!

**Comment by madig** [comment]
A clean-up of the outlines to make them interpolate cleanly would be preferable to a new master :)

**Comment by sladen** [comment]
There's two ways to go on this, the medium-term, and the long-term:
* *medium-term*; as @madig notes, fixing the number of order of the nodes created during the Fontlab cubic->quadratic conversion and overlap removal to be consistent across all weights.
* *long-term*; create a new "Ubuntu Extreme" fully interpolatable multi-axis multi-master super-magical variable font.  This will end up having slight differences, but would be Web 2.0 compliant and solve the general issue of new weight creation and the general over-heaviness of the Ubuntu family.

**Comment by madig** [comment]
Can quadratics be interpolated yet? And the VTT hinting code would probably have to be adjusted when changing point order around. Not sure how hard that is. Also, overlap removal needs to have resulted in the same number of nodes through all weights. Otherwise, changing the hinting becomes even hairier.

**Comment by anthrotype** [comment]
> the VTT hinting code would probably have to be adjusted when changing point order around

that's impossible right now. The "ubuntu extreme" will lose all the manual hinting.

> overlap removal needs to have resulted in the same number of nodes through all weights

that's also hard to accomplish. in fact, variable fonts are supposed to _not_ remove the overlaps, and renderers are supposed to support that.

**Comment by madig** [comment]
I was talking about the "medium-term" plan.

I personally favor the long-term one, which will probably require archaeology and restoration (the cubic outlines in the background have their overlaps removed and are therefore probably not the original outlines). I'd then leave hinting to ttfautomake and psautohint, should they work on variable fonts by then.

**Comment by anthrotype** [comment]
even the medium plan, how would you go about changing the quadratic points' order and number? by manually editing the q-curves it in a font editor?
And how would you remap the VTT instructions that are tied to the point indexes?
That's why I think the best way would be a completely new start, with cleaned up PS outlines and autohinting.

**Comment by sladen** [comment]
To create a *long-term* maintainable or "long-term" variable font, the manual hinting will ultimately have to die and be replaced with `ttfautohint`.

_(There was a huge time/money investment in manual hinting, so junking it is going to be a tough pill to swallow—but with each day, month and year that goes past the value of maintainability should become more appealing to all)._

**Comment by sladen** [comment]
_Opened daltonmaag/ubuntu/issues/25 as a placeholder for exploration of how to recover the overlapped cubics.

**Comment by madig** [comment]
> how would you go about changing the quadratic points' order and number?

No idea 😁  I didn't say I'm going to do it 😉  I'm all for starting fresh! The work I'm doing now is to simply get some momentum back into the project.

**Comment by anthrotype** [comment]
> The work I'm doing now is to simply get some momentum back into the project.

Thanks for this!
